Source/WebCore/inspector/agents/InspectorPageAgent.cpp

```diff
@@ -653,69 +656,76 @@ Inspector::Protocol::ErrorStringOr<Ref<JSON::ArrayOf<Inspector::Protocol::Page::
     return buildArrayForCookies(allRawCookies);
 }
 
-static std::optional<Cookie> parseCookieObject(Inspector::Protocol::ErrorString& errorString, Ref<JSON::Object>&& cookieObject)
+static std::pair<std::optional<Cookie>, IsPartitionedCookie> parseCookieObject(Inspector::Protocol::ErrorString& errorString, Ref<JSON::Object>&& cookieObject)
```

Comment:
could we reverse the order of this to be `std::optional<std::pair<Cookie, IsPartitionedCookie>>` so that we dont need `IsPartitionedCookie` if we're providing `std::nullopt`?